### DISCUSSION TITLE
error at stringify (<anonymous>)

### DISCUSSION THREAD

**Discussion by susahin80** [vercel/next.js#52830] [Help]
### Summary

I am creating a next app with the default options like this:

npx create-[next-app](https://linear.app/vercel/issue/NEXT-app)@latest
√ What is your project named? ... imdb
√ Would you like to use TypeScript? ... No / Yes
√ Would you like to use ESLint? ... No / Yes
√ Would you like to use Tailwind CSS? ... No / Yes
√ Would you like to use src/ directory? ... No / Yes
√ Would you like to use App Router? (recommended) ... No / Yes
√ Would you like to customize the default import alias? ... No / Yes

And after I start the app without making any changes with npm run dev and open localhost:3000 it gives the following error:

 npm run dev

> imdb@0.1.0 dev
> next dev

- ready started server on 0.0.0.0:3000, url: http://localhost:3000
- event compiled client and server successfully in 1573 ms (20 modules)
- wait compiling...
- event compiled client and server successfully in 129 ms (20 modules)
- wait compiling /page (client and server)...
- event compiled client and server successfully in 6s (474 modules)
- wait compiling...
- event compiled successfully in 125 ms (226 modules)
- error node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-server.edge.development.js (389:12) @ resolveClientReferenceMetadata
- error     at stringify (<anonymous>)
null
- error node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-server.edge.development.js (389:12) @ resolveClientReferenceMetadata
- error     at stringify (<anonymous>)
null
- error node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-server.edge.development.js (389:12) @ resolveClientReferenceMetadata
- error     at stringify (<anonymous>)
null
- error node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-server.edge.development.js (389:12) @ resolveClientReferenceMetadata
- error     at stringify (<anonymous>)
null
- error node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-server.edge.development.js (389:12) @ resolveClientReferenceMetadata
- error     at stringify (<anonymous>)
null
- error node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-server.edge.development.js (389:12) @ resolveClientReferenceMetadata
- error     at stringify (<anonymous>)
null
- error node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-server.edge.development.js (389:12) @ resolveClientReferenceMetadata
- error     at stringify (<anonymous>)
digest: "4023818524"
null

### Additional information

```js
package.json

  "dependencies": {
    "autoprefixer": "10.4.14",
    "eslint": "8.45.0",
    "eslint-config-next": "13.4.10",
    "next": "13.4.10",
    "postcss": "8.4.26",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "tailwindcss": "3.3.3"
  }
  
  node version: 18.16.1
```

**Comment by icyJoseph** [comment]
Could you make sure you have installed a Node version compatible with your OS architecture? Run `node -p 'process.arch'` - 

Also, in what OS is this happened? 

Could you try to use yarn instead? Just to see what happens?

**Reply by susahin80** [reply]
node -p 'process.arch'  returns x64.

And 

 npx --no-install next info

    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 18.16.1
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant Packages:
      next: 13.4.10
      eslint-config-next: 13.4.10
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
      
What is the exact command for yarn?

**Reply by icyJoseph** [reply]
You have to install yarn, like `npm i -g yarn` or something like that, look it up, then you do `yarn dev` on the next.js project. 

Could you also run `npm -g list --depth 0` - just want to make sure no global packages are affecting us.

Does the parent folder have node_modules installed?

**Reply by susahin80** [reply]
yarn dev gave the same error after I opened localhost:3000.

npm -g list --depth 0  shows the following:

C:\Users\susahin\AppData\Roaming\npm
+-- typescript@5.1.6
`-- yarn@1.22.19

**Reply by icyJoseph** [reply]
Alright, does the node architecture match the system's OS architecture? 

The parent directory, does not have any node_modules of its own right?

**Reply by susahin80** [reply]
I see "64-bit operating system, x64-based processor" in windows.
And node -p 'process.arch' returns x64. So I guess they match.

I have node_modules folder inside  C:\Users\susahin\AppData\Roaming\npm. And there are typescript and yarn inside it.

**Reply by icyJoseph** [reply]
Yeah but those are the globals, I thought that the parent folder of your project's location had node_modules.

I unfortunately have no access to Win at the moment, so I am not sure I can help you further.

Maybe, if you go inside the project, and run `npm explain react-dom` or `yarn why react-dom`

**Reply by susahin80** [reply]
Thank you for your time.
